culane_metric/evaluate.py

```diff
@@ -0,0 +1,205 @@
+""" Evaluation script for the CULane metric on the LLAMAS dataset.
+
+This script will compute the F1, precision and recall metrics as described in the CULane benchmark.
+
+The predictions format is the same one used in the CULane benchmark.
+In summary, for every annotation file:
+    labels/a/b/c.json
+There should be a prediction file:
+    predictions/a/b/c.lines.txt
+Inside each .lines.txt file each line will contain a sequence of points (x, y) separated by spaces.
+For more information, please see https://xingangpan.github.io/projects/CULane.html
+
+This script uses two methods to compute the IoU: one using an image to draw the lanes (named `discrete` here) and
+another one that uses shapes with the shapely library (named `continuous` here). The results achieved with the first
+method are very close to the official CULane implementation. Although the second should be a more exact method and is
+faster to compute, it deviates more from the official implementation. By default, the method closer to the official
+metric is used.
+"""
+
+import os
+import argparse
+from functools import partial
+
+import cv2
+import numpy as np
+from p_tqdm import t_map, p_map
+from scipy.interpolate import splprep, splev
+from scipy.optimize import linear_sum_assignment
+from shapely.geometry import LineString, Polygon
+
+import unsupervised_llamas.common.helper_scripts as helper_scripts
+import unsupervised_llamas.label_scripts.spline_creator as spline_creator
+
+LLAMAS_IMG_RES = (717, 1276)
```

Comment:
Optional: The size constants are stored within common/constants.  It would be cleaner to have them defined only once